dask_ml/wrappers.py

```diff
@@ -279,10 +279,15 @@ def predict(self, X):
             return result
 
         elif isinstance(X, dd._Frame):
+            try:
+                meta = _predict(X._meta_nonempty, estimator=self._postfit_estimator)
+            except Exception:
```

Comment:
This would generally handle all cases though I'm not sure when it would come up.  @TomAugspurger do you have any better ideas on how to handle ?

Reply:
Depending upon the model we can have various errors.   `IndexErrors` is one such example with  `CategoricalNB` model) (Added a test for it).

Sorry if you were all-ready aware but added a comment in detail [here](https://github.com/dask/dask-ml/pull/862#issuecomment-951341770).

Reply:
I don't really see the need for this try / except

I'd prefer to handle `meta` here similarly to how we handle it everywhere in Dask: the default should be `None` (infer) and users should be able to override it. 

The wrinkle here is that `ParallelPostFit` previously assumed the output was int, so we need to deprecate that behavior.

So IMO we add a new keyword to `ParallelPostFit` called `meta` that gets passed here. By default it's `None`, which means warn and set it equal to `np.array([1])`. But in the future it'll just be passed through as is to `map_partitions`. Does that make sense?

Reply:
> So IMO we add a new keyword to ParallelPostFit called meta that gets passed here. By default it's None, which means warn and set it equal to np.array([1]). But in the future it'll just be passed through as is to map_partitions. Does that make sense?

Gotcha. Thanks for the explanation.  Sorry for the back and forth. Just to be clear the plan is as follows : 

1. For the curent release, add a `meta` keyword which falls back to `np.array([1])` and raises warnings if not provided in the call. 

2. For future releases deprecate the `np.array([1])` fallback behavior by passing `meta` as is . 


Happy to update this PR to reflect `1` and raise an issue to track `2`  if we are on the same page regarding this.

Reply:
@TomAugspurger did you have any thoughts on Vibhu's proposal above? 🙂

Reply:
Yep that sounds right.

Reply:
Thanks @TomAugspurger . Pushing on an update right away.

Reply:
Pushed the requested changed  to the PR  commit: ( https://github.com/dask/dask-ml/pull/862/commits/f6316a978a066a6c2ebdea7dbee4da4a9b9393e8 )

Can i get a review on this. Raising an issue for 2 now.

Reply:
Added issue as to track 2 [here](https://github.com/dask/dask-ml/pull/862). Please let me know if there is anything else that needs to be done to push this through.